tests/test_pyglmnet.py

```diff
@@ -388,3 +385,61 @@ def test_random_state_consistency():
 
     assert_array_equal(ypred_a, ypred_b)
     assert_array_equal(ypred_b, ypred_c)
+
+
+@pytest.mark.parametrize("distr", ALLOWED_DISTRS)
+def test_simulate_glm(distr):
+    """
+    Test that every generative model can be simulated from.
+    """
+
+    n_samples, n_features = 10, 3
+
+    # sample random coefficients
+    beta0 = np.random.normal(0.0, 1.0, 1)
+    beta = np.random.normal(0.0, 1.0, n_features)
+
+    np.random.seed(1)
+    X = np.random.normal(0.0, 1.0, [n_samples, n_features])
+    simulate_glm(distr, beta0, beta, X)
+
+
+def test_simulate_glm_failing():
+    """
+    Test that an error is raised if an invalid parameter name is passed.
+    """
+    distr = 'multivariate_gaussian_poisson'
+    with pytest.raises(ValueError):
+        simulate_glm(distr, 1.0, 1.0, np.array([[1.0]]))
+
+
+@pytest.mark.parametrize("y_func",
+                         [list, tuple, np.array,
+                          lambda x: np.array(x).reshape(-1, 1)])
+def test_API_input_types_y(y_func):
+    """
+    Test that the input value of y can be of different types.
+    """
+    np.random.seed(1)
+    n_samples, n_features = 100, 5
+
+    X = np.random.randn(n_samples, n_features)
+    y = np.random.randn(n_samples)
+
+    y = y_func(y)
+
+    glm = GLM(distr='gaussian')
+    glm.fit(X, y)
+
+    glm.predict(X)
+    glm.score(X, y)
```

Comment:
just a smoke test. I would suggest checking if errors are raised when it's not the right dimensions